src/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/SymmetricKeyWrap.cs

```diff
@@ -62,31 +66,33 @@ internal static byte[] TripleDESKeyWrapDecrypt(byte[] rgbKey, byte[] rgbEncrypte
                 && rgbEncryptedWrappedKeyData.Length != 48)
                 throw new CryptographicException(SR.Cryptography_Xml_KW_BadKeySize);
 
-            TripleDESCryptoServiceProvider tripleDES = new TripleDESCryptoServiceProvider();
-            // Assume no padding, use CBC mode
-            tripleDES.Padding = PaddingMode.None;
-            ICryptoTransform dec1 = tripleDES.CreateDecryptor(rgbKey, s_rgbTripleDES_KW_IV);
-            byte[] temp2 = dec1.TransformFinalBlock(rgbEncryptedWrappedKeyData, 0, rgbEncryptedWrappedKeyData.Length);
-            Array.Reverse(temp2);
-            // Get the IV and temp1
-            byte[] rgbIV = new byte[8];
-            Buffer.BlockCopy(temp2, 0, rgbIV, 0, 8);
-            byte[] temp1 = new byte[temp2.Length - rgbIV.Length];
-            Buffer.BlockCopy(temp2, 8, temp1, 0, temp1.Length);
-
-            ICryptoTransform dec2 = tripleDES.CreateDecryptor(rgbKey, rgbIV);
-            byte[] rgbWKCKS = dec2.TransformFinalBlock(temp1, 0, temp1.Length);
-
-            // checksum the key
-            byte[] rgbWrappedKeyData = new byte[rgbWKCKS.Length - 8];
-            Buffer.BlockCopy(rgbWKCKS, 0, rgbWrappedKeyData, 0, rgbWrappedKeyData.Length);
-            using (var sha = SHA1.Create())
+            using (TripleDES tripleDES = TripleDES.Create())
             {
-                byte[] rgbCKS = sha.ComputeHash(rgbWrappedKeyData);
-                for (int index = rgbWrappedKeyData.Length, index1 = 0; index < rgbWKCKS.Length; index++, index1++)
-                    if (rgbWKCKS[index] != rgbCKS[index1])
-                        throw new CryptographicException(SR.Cryptography_Xml_BadWrappedKeySize);
-                return rgbWrappedKeyData;
+                // Assume no padding, use CBC mode
+                tripleDES.Padding = PaddingMode.None;
+                ICryptoTransform dec1 = tripleDES.CreateDecryptor(rgbKey, s_rgbTripleDES_KW_IV);
```

Comment:
Sorry, my mistake. We also have RijndaelManaged and ICryptoTransform lekages in AESKeyWrapEncrypt / AESKeyWrapDecrypt, will fix them as well